### PR TITLE
Skip new params for old driver compiler

### PR DESCRIPTION
**Pull request by XinWangIntel**
### Details:
 There are new added options for NPU_COMPILATION_MODE_PARAMS, need to skip for old driver compiler.

### Tickets:
 - *146197*

### CODE REVIEW

File: src/inference/include/openvino/runtime/intel_npu/properties.hpp [hunk 1]
```diff
@@ -53,5 +53,13 @@ static constexpr ov::Property<uint64_t, ov::PropertyMutability::RO> device_total
  */
 static constexpr ov::Property<uint32_t, ov::PropertyMutability::RO> driver_version{"NPU_DRIVER_VERSION"};
 
+/**
+ * @brief [Only for NPU compiler]
+ * Type: std::string
+ * Set various parameters supported by the NPU compiler.
+ * @ingroup ov_runtime_npu_prop_cpp_api
+ */
+static constexpr ov::Property<std::string> compilation_mode_params{"NPU_COMPILATION_MODE_PARAMS"};
```
Comment (ArtemySkrebkov) [inline src/inference/include/openvino/runtime/intel_npu/properties.hpp:62]: There is also an method `isPublic`, it is set to false for now:

https://github.com/openvinotoolkit/openvino/blob/84530e7f33ddc66ebbc9912319f47502b268a852/src/plugins/intel_npu/src/al/include/intel_npu/al/config/compiler.hpp#L159

However, I am not sure if the method is used in the current implementation

Comment (XinWangIntel) [reply src/inference/include/openvino/runtime/intel_npu/properties.hpp:62]: Changed to TRUE now, Thanks!

**Review by MirceaDan99 — APPROVED** [review]

**Review by ArtemySkrebkov — APPROVED** [review]

File: src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp [hunk 2]
```diff
@@ -376,6 +376,27 @@ std::string LevelZeroCompilerInDriver<TableExtension>::serializeConfig(
...
     std::string content = config.toString();
+    _logger.debug("Original content of config: %s", content.c_str());
+
+    // Remove optimization-level and performance-hint-override for old driver which not support them
+    if ((compilerVersion.major < 5) || (compilerVersion.major == 5 && compilerVersion.minor < 7)) {
+        std::ostringstream optLevelStr;
+        optLevelStr << "optimization-level" << KEY_VALUE_SEPARATOR << "\\d+";
+        std::ostringstream perfHintStr;
+        perfHintStr << "performance-hint-override" << KEY_VALUE_SEPARATOR << "\\S+";
+        _logger.warning(
```
Comment (nikita-kud) [inline src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp:387]: Should not we first check if `ptimization-level`/`performance-hint-override` are present?

```
auto areOptionsSupported = (compilerVersion.major > 5) || (compilerVersion.major == 5 && compilerVersion.minor > 7);
auto contaisOptions = content.contains(optLevelStr) || content.contains(perfHintStr); // pseudo-code

if (!areOptionsSupported  && contaisOptions) { 
// write message
// remove options
}
```

Comment (XinWangIntel) [reply src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp]: Added more strict check now, please help to review again

**Pull request merged**